python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -142,6 +145,27 @@ def compute_intersection_with_partition_keys(
         """Return a new AssetSlice with only the given partition keys if they are in the slice."""
         return self._asset_graph_view.compute_intersection_with_partition_keys(partition_keys, self)
 
+    def __or__(self, other: Union["AssetSlice", "AssetSubset"]) -> "AssetSlice":
+        if isinstance(other, AssetSlice):
+            return _slice_from_subset(
+                self._asset_graph_view, self._compatible_subset | other._compatible_subset
+            )
+        return _slice_from_subset(self._asset_graph_view, self._compatible_subset | other)
+
+    def __sub__(self, other: Union["AssetSlice", "AssetSubset"]) -> "AssetSlice":
+        if isinstance(other, AssetSlice):
+            return _slice_from_subset(
+                self._asset_graph_view, self._compatible_subset - other._compatible_subset
+            )
+        return _slice_from_subset(self._asset_graph_view, self._compatible_subset - other)
+
+    def __and__(self, other: Union["AssetSlice", "AssetSubset"]) -> "AssetSlice":
+        if isinstance(other, AssetSlice):
+            return _slice_from_subset(
+                self._asset_graph_view, self._compatible_subset & other._compatible_subset
+            )
+        return _slice_from_subset(self._asset_graph_view, self._compatible_subset & other)
```

Comment:
this is just an RFC but just a heads up that I will push back against operator overloading in this context, and especially against mixed types

Reply:
I don't mind doing `.intersect()` and `.union()` etc. if that makes things feel better, just certainly need some way of doing these operations